### ISSUE TITLE
Issue with local test deployment: upstream prematurely closed connection while reading response header from upstream

### ISSUE THREAD

**Issue by JaTat** [frappe/frappe_docker#257]
Hey all,
I am new and experiencing some difficulties with getting erpnext running with docker-compose. 

My setup:
Win 10, docker-compose 1.25.5

I am following https://github.com/frappe/frappe_docker and want to start the local deployment for testing, i.e. run expose on localhost.

My .env:
```
    VERSION=version-12 (I also tried edge)
    MARIADB_HOST=mariadb
    MYSQL_ROOT_PASSWORD=admin
    SITES=test.localhost
    LETSENCRYPT_EMAIL=<my_mail>
```

I can docker-compose up all the containers according to the readme:

```
docker-compose \
    --project-name <project-name> \
    -f installation/docker-compose-common.yml \
    -f installation/docker-compose-erpnext.yml \
    -f installation/erpnext-publish.yml \
    up -d
```

But when I open localhost:80 in browser I get the _sorry we will be back soon error page_. I get no errors from docker-compose -f … logs,
**but: docker logs erpnext-nginx_1 gives me:**

```
error] 29#29: *27 upstream prematurely closed connection  
while reading response header from upstream, client:
172.18.0.1, server: $http_host, request: "GET / HTTP/1.1",
upstream: "http://172.18.0.5:8000/", host: "127.0.0.1"
172.18.0.1 - - [11/Jun/2020:18:49:12 +0000] "GET / HTTP/1.1"
502 5444 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64;
rv:77.0) Gecko/20100101 Firefox/77.0" "-"
```

172.18.0.5:8000 is the python worker erpnext-python_1 and a curl -v 172.18.0.5:8000 from one of the containers i get an empty reply.

Am i missing something? 

I also tried to setup a new site with 

```
   docker run `
    -e "SITE_NAME=site2.localhost" `
    -e "DB_ROOT_USER=root" `
    -e "MYSQL_ROOT_PASSWORD=admin" `
    -e "ADMIN_PASSWORD=admin" `
    -e "INSTALL_APPS=erpnext" `
    -v JanERP_sites-vol:/home/frappe/frappe-bench/sites `
    --network janerp_default `
    frappe/erpnext-worker:version-12 new
```

I get an error with quite some traceback starting in the new.py main() it tries to setup the mariadb connection eventually leading to:

```
  File "/home/frappe/frappe-bench/env/lib/python3.7/site-packages/pymysql/connections.py", line 630, in connect
    raise exc
pymysql.err.OperationalError: (2003, "Can't connect to MySQL server on 'localhost' ([Errno 99] Cannot assign requested address)")
```

Thanks in advance

**Comment by revant** [comment]
I don't have windows 10 to try this.

Once I faced something on windows about Mariadb 

Just to note about windows

- Set environment variable COMPOSE_CONVERT_WINDOWS_PATHS e.g. set COMPOSE_CONVERT_WINDOWS_PATHS=1
- Make the frappe-mariadb.cnf read-only for mariadb container to pick it up.

**Comment by JaTat** [comment]
Thank you @revant for the quick reply. I set the environment variable and write protected the frappe-mariadb.cnf. Then I pruned all images and rebuilt.
When starting a new site it the config checker in the docker-entrypoint.sh could not find the  common_site_config.json. I assumed thats because of the write protection so I did another attempt without write protecting the frappe-mariadb.cnf.

Now I the error when creating a new site is gone. It installs frappe and erpnext and finishes with **Updating customizations for Address
*** Scheduler is disabled *****

I still get the same error in the nginx log as above and the Sorry! we will be back soon site.

I think I will just try it on a Linux VM.

Best and many thanks
